packages/aws-cdk-lib/aws-opensearchservice/test/domain.test.ts

```diff
@@ -1789,6 +1795,88 @@ each(testedOpenSearchVersions).describe('custom endpoints', (engineVersion) => {
   });
 });
 
+each(testedOpenSearchVersions).describe('TLS security policy', (engineVersion) => {
+  test('defaults to TLS 1.2 when tlsSecurityPolicy is not specified', () => {
+    new Domain(stack, 'Domain', {
+      version: engineVersion,
+    });
+
+    Template.fromStack(stack).hasResourceProperties('AWS::OpenSearchService::Domain', {
+      DomainEndpointOptions: {
+        TLSSecurityPolicy: 'Policy-Min-TLS-1-2-2019-07',
+      },
+    });
+  });
+
+  test('defaults to TLS 1.2 when enforceHttps is true but tlsSecurityPolicy is not specified', () => {
+    new Domain(stack, 'Domain', {
+      version: engineVersion,
+      enforceHttps: true,
+    });
+
+    Template.fromStack(stack).hasResourceProperties('AWS::OpenSearchService::Domain', {
+      DomainEndpointOptions: {
+        EnforceHTTPS: true,
+        TLSSecurityPolicy: 'Policy-Min-TLS-1-2-2019-07',
+      },
+    });
+  });
+
+  test('uses TLS 1.0 when explicitly specified', () => {
+    new Domain(stack, 'Domain', {
+      version: engineVersion,
+      tlsSecurityPolicy: TLSSecurityPolicy.TLS_1_0,
+    });
+
+    Template.fromStack(stack).hasResourceProperties('AWS::OpenSearchService::Domain', {
+      DomainEndpointOptions: {
+        TLSSecurityPolicy: 'Policy-Min-TLS-1-0-2019-07',
+      },
+    });
+  });
+
+  test('uses TLS 1.2 when explicitly specified', () => {
+    new Domain(stack, 'Domain', {
+      version: engineVersion,
+      tlsSecurityPolicy: TLSSecurityPolicy.TLS_1_2,
+    });
+
+    Template.fromStack(stack).hasResourceProperties('AWS::OpenSearchService::Domain', {
+      DomainEndpointOptions: {
+        TLSSecurityPolicy: 'Policy-Min-TLS-1-2-2019-07',
+      },
+    });
+  });
+
+  test('uses TLS 1.2 PFS when explicitly specified', () => {
+    new Domain(stack, 'Domain', {
+      version: engineVersion,
+      tlsSecurityPolicy: TLSSecurityPolicy.TLS_1_2_PFS,
+    });
+
+    Template.fromStack(stack).hasResourceProperties('AWS::OpenSearchService::Domain', {
+      DomainEndpointOptions: {
+        TLSSecurityPolicy: 'Policy-Min-TLS-1-2-PFS-2023-10',
+      },
+    });
+  });
+
+  test('respects explicit tlsSecurityPolicy even when enforceHttps is enabled', () => {
```

Comment:
Is there some logic in the code which ensures this? Maybe I am missing it, but I do not see any conditional checks involving `enforceHttps`